src/ClientAssets/Microsoft.AspNetCore.ClientAssets/build/netstandard2.0/Microsoft.AspNetCore.ClientAssets.targets

```
@@ -32,7 +32,7 @@
     </PropertyGroup>
 
     <MakeDir Directories="$(_ClientAssetsOutputFullPath)" />
-    <Exec Command="$(ClientAssetsBuildCommand) -- $(ClientAssetsBuildOutputParameter) $(_ClientAssetsOutputFullPath)" WorkingDirectory="$(ClientAssetsDirectory)" />
+    <Exec Command="$(ClientAssetsBuildCommand) -- $(ClientAssetsBuildOutputParameter) &quot;$(_ClientAssetsOutputFullPath)&quot;" WorkingDirectory="$(ClientAssetsDirectory)" />
```

Does the following work?

```suggestion
    <Exec Command="$(ClientAssetsBuildCommand) -- $(ClientAssetsBuildOutputParameter) '$(_ClientAssetsOutputFullPath)'" WorkingDirectory="$(ClientAssetsDirectory)" />
```

---

For some reason, it wants the ``quot;`` instead of a real quote.  because a single quote will again result in an error.

I am not sure if this is a webpack "thing", but I must admit the fact that my "solution" works is odd.

I prepared a minimal repro https://github.com/MariovanZeist/AssetsInSpace to show the problem.

Try building this project from a folder without spaces in them, and again with spaces in them and see the difference.

---

@TanayParikh I did a little more research and found that besides using ``quot; `` escaping with ``%22`` also works.
The closest to official MS documentation on this I got was https://docs.microsoft.com/en-us/visualstudio/msbuild/special-characters-to-escape?view=vs-2022 and especially the note on that page stating:
> In some scenarios, you may need to escape double quote (") characters, such as when using within an Exec task

It's not actually ``MSBuild`` that is failing, but the executable ``npm`` that it called with the ``Exec`` task. It interpreted spaces as separate commandos.

---

@TanayParikh I was under the mistaken assumption this package would be used for production, but @javiercn cleared it up for me https://github.com/dotnet/aspnetcore/issues/38445#issuecomment-1196549022

So for me, this problem is resolved.